update mysql to 22.1.0

Hi and thank you for putting the efforts to make this repo.

I'm trying to use it with mysql but I'm getting an error.

`Cargo.toml`
```toml
[package]
name = "db"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
mysql_async = "0.29.0"
mysql = "22.0.0"
tokio = { version = "1.0", features = ["macros", "rt-multi-thread"] }
refinery = { version="0.8.4", features = ["mysql"] }
serde_json = "1.0.79"
```

`main.rs`

```rust
mod embedded {
    use refinery::embed_migrations;
    embed_migrations!("migrations");
}

fn main() {
    use mysql::Opts;

    let url = "mysql://root:pass1234@localhost:3306";
    let mut conn = mysql::Conn::new(Opts::from_url(url).unwrap()).unwrap();
    embedded::migrations::runner().run(&mut conn).unwrap();
}
```


`result`

```bash
error[E0277]: the trait bound `Conn: Migrate` is not satisfied
   --> db\src\bin\migrations.rs:25:40
    |
25  |     embedded::migrations::runner().run(&mut conn).unwrap();
    |                                    --- ^^^^^^^^^ the trait `Migrate` is not implemented for `Conn`
    |                                    |
    |                                    required by a bound introduced by this call
    |
note: required by a bound in `Runner::run`
   --> C:\Users\arkan\.cargo\registry\src\github.com-1ecc6299db9ec823\refinery-core-0.8.4\src\runner.rs:371:12
    |
371 |         C: Migrate,
    |            ^^^^^^^ required by this bound in `Runner::run`

For more information about this error, try `rustc --explain E0277`.


```

@pintariching I've removed `mysql`, then I ran `cargo clean` then `cargo build`

`migrations.rs`
```rust
mod embedded {
    use refinery::embed_migrations;
    embed_migrations!("migrations");
}

fn main() {
    use mysql::Opts;

    let url = "mysql://root:pass1234@localhost:3306";
    let mut conn = mysql::Conn::new(Opts::from_url(url).unwrap()).unwrap();
    embedded::migrations::runner().run(&mut conn).unwrap();
}
```


`Cargo.toml`
```toml
[package]
name = "db"
version = "0.1.0"
edition = "2021"

[dependencies]
mysql = "22.0.0"
tokio = { version = "1.0", features = ["macros", "rt-multi-thread"] }
refinery = { version="0.8.4", features = ["mysql"] }
serde_json = "1.0.79"
```
`cargo clean`

`cargo build`
```shell
   Compiling proc-macro2 v1.0.36
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.86
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
   Compiling autocfg v1.1.0
   Compiling winapi v0.3.9
   Compiling memchr v2.4.1
   Compiling lazy_static v1.4.0
   Compiling log v0.4.14
   Compiling libc v0.2.119
   Compiling bytes v1.1.0
   Compiling serde_derive v1.0.136
   Compiling cc v1.0.73
   Compiling serde v1.0.136
   Compiling once_cell v1.9.0
   Compiling pin-project-lite v0.2.8
   Compiling itoa v1.0.1
   Compiling typenum v1.15.0
   Compiling ntapi v0.3.7
   Compiling regex-syntax v0.6.25
   Compiling futures-core v0.3.21
   Compiling ppv-lite86 v0.2.16
   Compiling ryu v1.0.9
   Compiling num_cpus v1.13.1
   Compiling pkg-config v0.3.24
   Compiling vcpkg v0.2.15
   Compiling crc32fast v1.3.2
   Compiling serde_json v1.0.79
   Compiling base64 v0.13.0
   Compiling futures-sink v0.3.21
   Compiling percent-encoding v2.1.0
   Compiling matches v0.1.9
   Compiling tinyvec_macros v0.1.0
   Compiling native-tls v0.2.8
   Compiling futures-task v0.3.21
   Compiling futures-channel v0.3.21
   Compiling futures-util v0.3.21
   Compiling byteorder v1.4.3
   Compiling unicode-bidi v0.3.7
   Compiling futures-io v0.3.21
   Compiling slab v0.4.5
   Compiling glob v0.3.0
   Compiling pin-utils v0.1.0
   Compiling static_assertions v1.1.0
   Compiling bitflags v1.3.2
   Compiling proc-macro-hack v0.5.19
   Compiling fnv v1.0.7
   Compiling unicode-segmentation v1.9.0
   Compiling either v1.6.1
   Compiling httparse v1.6.0
   Compiling tower-service v0.3.1
   Compiling getrandom v0.1.16
   Compiling shlex v1.1.0
   Compiling try-lock v0.2.3
   Compiling lazycell v1.3.0
   Compiling peeking_take_while v0.1.2
   Compiling rustc-hash v1.1.0
   Compiling frunk_core v0.4.0
   Compiling httpdate v1.0.2
   Compiling anyhow v1.0.55
   Compiling async-trait v0.1.52
   Compiling unicode-width v0.1.9
   Compiling encoding_rs v0.8.30
   Compiling failure_derive v0.1.8
   Compiling adler v1.0.2
   Compiling rle-decode-fast v1.0.3
   Compiling cpufeatures v0.2.1
   Compiling gimli v0.26.1
   Compiling radium v0.6.2
   Compiling rust_decimal v1.22.0
   Compiling strsim v0.8.0
   Compiling vec_map v0.8.2
   Compiling bindgen v0.58.1
   Compiling humantime v2.1.0
   Compiling rustc-demangle v0.1.21
   Compiling adler32 v1.2.0
   Compiling tap v1.0.1
   Compiling arrayvec v0.7.2
   Compiling strum v0.18.0
   Compiling lexical-core v0.7.6
   Compiling crossbeam-utils v0.8.7
   Compiling ipnet v2.3.1
   Compiling funty v1.2.0
   Compiling const_fn v0.4.9
   Compiling mime v0.3.16
   Compiling minimal-lexical v0.2.1
   Compiling arrayvec v0.5.2
   Compiling fastrand v1.7.0
   Compiling smallvec v1.8.0
   Compiling fixedbitset v0.4.1
   Compiling saturating v0.1.0
   Compiling multimap v0.8.3
   Compiling opaque-debug v0.3.0
   Compiling bindgen v0.59.2
   Compiling sha1_smol v1.0.0
   Compiling bufstream v0.1.4
   Compiling crossbeam-epoch v0.9.7
   Compiling mysql v21.0.2
   Compiling scopeguard v1.1.0
   Compiling crossbeam-queue v0.3.4
   Compiling tower-layer v0.3.1
   Compiling siphasher v0.3.9
   Compiling mysql v22.1.0
   Compiling getrandom v0.2.5
   Compiling tracing-core v0.1.22
   Compiling time v0.3.7
   Compiling ahash v0.7.6
   Compiling generic-array v0.14.5
   Compiling nom v5.1.2
   Compiling standback v0.2.17
   Compiling nom v7.1.0
   Compiling time v0.2.27
   Compiling tinyvec v1.5.1
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling indexmap v1.8.0
   Compiling miniz_oxide v0.4.4
   Compiling num-bigint v0.4.3
   Compiling num-bigint v0.2.6
   Compiling num-bigint v0.3.3
   Compiling io-enum v1.0.1
   Compiling memoffset v0.6.5
   Compiling form_urlencoded v1.0.1
   Compiling cmake v0.1.48
   Compiling pem v1.0.2
   Compiling lexical-util v0.8.1
   Compiling http v0.2.6
   Compiling itertools v0.10.3
   Compiling textwrap v0.11.0
   Compiling libflate_lz77 v1.1.0
   Compiling heck v0.3.3
   Compiling wyz v0.4.0
   Compiling clang-sys v1.3.1
   Compiling sha1 v0.6.1
   Compiling rand_core v0.6.3
   Compiling libz-sys v1.1.3
   Compiling backtrace v0.3.64
   Compiling unicode-normalization v0.1.19
   Compiling rdkafka-sys v4.2.0+1.8.2
   Compiling addr2line v0.17.0
   Compiling lexical-parse-integer v0.8.0
   Compiling lexical-write-integer v0.8.0
   Compiling want v0.3.0
   Compiling rand_chacha v0.3.1
   Compiling quote v1.0.15
   Compiling which v4.2.4
   Compiling which v3.1.1
   Compiling aho-corasick v0.7.18
   Compiling object v0.27.1
   Compiling libflate v1.1.2
   Compiling http-body v0.4.4
   Compiling rand_core v0.5.1
   Compiling lexical-parse-float v0.8.2
   Compiling lexical-write-float v0.8.2
   Compiling bitvec v0.22.3
   Compiling rand v0.8.5
   Compiling hashbrown v0.11.2
   Compiling crossbeam-channel v0.5.2
   Compiling prost-build v0.9.0
   Compiling rand_chacha v0.2.2
   Compiling idna v0.2.3
   Compiling rand v0.7.3
   Compiling crossbeam-deque v0.8.1
   Compiling lexical-core v0.8.2
   Compiling regex v1.5.4
   Compiling lru v0.6.6
   Compiling lru v0.7.2
   Compiling crossbeam v0.8.1
   Compiling lexical v6.0.1
   Compiling url v2.2.2
   Compiling twox-hash v1.6.2
   Compiling flate2 v1.0.22
   Compiling digest v0.9.0
   Compiling block-buffer v0.9.0
   Compiling block-buffer v0.10.2
   Compiling crypto-common v0.1.3
   Compiling petgraph v0.6.0
   Compiling sha2 v0.9.9
   Compiling digest v0.10.3
   Compiling lexical v5.2.2
   Compiling sha-1 v0.10.0
   Compiling sha2 v0.10.2
   Compiling cexpr v0.4.0
   Compiling pem v0.8.3
   Compiling cexpr v0.6.0
   Compiling bigdecimal v0.3.0
   Compiling socket2 v0.4.4
   Compiling miow v0.3.7
   Compiling schannel v0.1.19
   Compiling libloading v0.7.3
   Compiling winapi-util v0.1.5
   Compiling subprocess v0.2.8
   Compiling atty v0.2.14
   Compiling winreg v0.7.0
   Compiling remove_dir_all v0.5.3
   Compiling time v0.1.44
   Compiling named_pipe v0.4.1
   Compiling synstructure v0.12.6
   Compiling frunk_proc_macro_helpers v0.1.0
   Compiling derive_utils v0.11.2
   Compiling clap v2.34.0
   Compiling termcolor v1.1.2
   Compiling same-file v1.0.6
   Compiling tempfile v3.3.0
   Compiling walkdir v2.3.2
   Compiling env_logger v0.8.4
   Compiling tokio-macros v1.7.0
   Compiling thiserror-impl v1.0.30
   Compiling futures-macro v0.3.21
   Compiling tracing-attributes v0.1.19
   Compiling zerocopy-derive v0.2.0
   Compiling prost-derive v0.9.0
   Compiling frunk_proc_macros_impl v0.1.0
   Compiling strum_macros v0.18.0
   Compiling typed-builder v0.5.1
   Compiling frunk_derives v0.4.0
   Compiling time-macros-impl v0.1.2
   Compiling pin-project-internal v1.0.10
   Compiling async-stream-impl v0.3.2
   Compiling mio v0.8.0
   Compiling frunk_proc_macros v0.1.0
   Compiling failure v0.1.8
   Compiling tokio v1.17.0
   Compiling time-macros v0.1.1
   Compiling frunk v0.4.0
   Compiling async-stream v0.3.2
   Compiling zerocopy v0.3.0
   Compiling thiserror v1.0.30
   Compiling pin-project v1.0.10
   Compiling tracing v0.1.31
   Compiling prost v0.9.0
   Compiling tracing-futures v0.2.5
   Compiling prost-types v0.9.0
   Compiling tonic-build v0.6.2
   Compiling futures-executor v0.3.21
   Compiling tokio-util v0.6.9
   Compiling tokio-native-tls v0.3.0
   Compiling tokio-io-timeout v1.2.0
   Compiling tokio-util v0.7.0
   Compiling tokio-stream v0.1.8
   Compiling futures v0.3.21
   Compiling tower v0.4.12
   Compiling h2 v0.3.11
   Compiling id_and_access v0.1.0 (C:\Users\arkan\dev\projects\esmartbill\id_and_access\id_and_access)
   Compiling mysql_common v0.28.2
   Compiling mysql_common v0.27.5
   Compiling uuid v0.8.2
   Compiling toml v0.5.8
   Compiling serde_urlencoded v0.7.1
   Compiling bigdecimal v0.2.2
   Compiling chrono v0.4.19
   Compiling avro-rs v0.13.0
   Compiling proc-macro-crate v1.1.3
   Compiling num_enum_derive v0.5.6
   Compiling num_enum v0.5.6
   Compiling hyper v0.14.17
   Compiling refinery-core v0.8.4
   Compiling refinery-macros v0.8.4
   Compiling refinery v0.8.4
   Compiling db v0.1.0 (C:\Users\arkan\dev\projects\esmartbill\id_and_access\db)
   Compiling hyper-tls v0.5.0
   Compiling hyper-timeout v0.4.1
error[E0277]: the trait bound `Conn: Migrate` is not satisfied
   --> db\src\bin\migrations.rs:25:40
    |
25  |     embedded::migrations::runner().run(&mut conn).unwrap();
    |                                    --- ^^^^^^^^^ the trait `Migrate` is not implemented for `Conn`
    |                                    |
    |                                    required by a bound introduced by this call
    |
note: required by a bound in `Runner::run`
   --> C:\Users\arkan\.cargo\registry\src\github.com-1ecc6299db9ec823\refinery-core-0.8.4\src\runner.rs:371:12
    |
371 |         C: Migrate,
    |            ^^^^^^^ required by this bound in `Runner::run`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `db` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Hi @jxs 
I tried your suggestion to use `mysql="=22.0.0"` but it is giving the same error.

`Cargo.toml`
```toml
[package]
name = "db"
version = "0.1.0"
edition = "2021"

[dependencies]
mysql = "=22.0.0"
tokio = { version = "1.0", features = ["macros", "rt-multi-thread"] }
refinery = { version="0.8.4", features = ["mysql"] }
serde_json = "1.0.79"
```

`cargo build`

```shell
   Compiling ...
   Compiling schema_registry_converter v2.1.0
   Compiling db v0.1.0 (C:\Users\arkan\dev\projects\esmartbill\id_and_access\db)
error[E0277]: the trait bound `Conn: Migrate` is not satisfied
   --> db\src\bin\migrations.rs:25:40
    |
25  |     embedded::migrations::runner().run(&mut conn).unwrap();
    |                                    --- ^^^^^^^^^ the trait `Migrate` is not implemented for `Conn`
    |                                    |
    |                                    required by a bound introduced by this call
    |
note: required by a bound in `Runner::run`
   --> C:\Users\arkan\.cargo\registry\src\github.com-1ecc6299db9ec823\refinery-core-0.8.4\src\runner.rs:371:12
    |
371 |         C: Migrate,
    |            ^^^^^^^ required by this bound in `Runner::run`

For more information about this error, try `rustc --explain E0277`.
   Compiling rdkafka v0.28.0
error: could not compile `db` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Thank you

--- a/refinery_core/Cargo.toml
+++ b/refinery_core/Cargo.toml
@@ -33,7 +33,7 @@ walkdir = "2.3.1"
 rusqlite = { version = ">= 0.23, <= 0.26", optional = true }
 postgres = { version = "0.19", optional = true }
 tokio-postgres-driver = { package = "tokio-postgres", version = "0.7", optional = true }
-mysql = { version = ">= 21.0.0, <= 22.0.0", optional = true, default-features = false}
+mysql = { version = ">= 21.0.0, <= 22", optional = true, default-features = false}
 mysql-async-driver = { package = "mysql_async", version = ">= 0.28, <= 0.29", optional = true }
 tokio = { version = "1.0", features = ["full"], optional = true }
 tiberius-driver = { package = "tiberius", version = "0.6", optional = true }

